testing/ef-tests/src/cases/blockchain_test.rs

```diff
@@ -64,93 +65,107 @@ impl Case for BlockchainTestCase {
+                    let decoded = SealedBlock::decode(&mut block.rlp.as_ref())?;
                     provider
-                        .insert_hashes(
-                            0..=last_block.number,
-                            last_block.hash(),
-                            *expected_state_root,
+                        .insert_historical_block(
+                            decoded.clone().try_seal_with_senders().unwrap(),
+                            None,
                         )
                         .map_err(|err| Error::RethError(err.into()))?;
+                    Ok::<Option<SealedBlock>, Error>(Some(decoded))
+                })?;
+                provider
+                    .static_file_provider()
+                    .latest_writer(StaticFileSegment::Headers)
+                    .unwrap()
+                    .commit()
+                    .unwrap();
+
+                // Execute the execution stage using the EVM processor factory for the test case
+                // network.
+                let _ = ExecutionStage::new_with_factory(reth_revm::EvmProcessorFactory::new(
+                    Arc::new(case.network.clone().into()),
+                    EthEvmConfig::default(),
+                ))
+                .execute(
+                    &provider,
+                    ExecInput { target: last_block.as_ref().map(|b| b.number), checkpoint: None },
+                );
+
+                // Validate the post-state for the test case.
+                match (&case.post_state, &case.post_state_hash) {
+                    (Some(state), None) => {
+                        // Validate accounts in the state against the provider's database.
+                        for (&address, account) in state.iter() {
+                            account.assert_db(address, provider.tx_ref())?;
+                        }
+                    }
+                    (None, Some(expected_state_root)) => {
+                        // Insert state hashes into the provider based on the expected state root.
+                        let last_block = last_block.unwrap_or_default();
+                        provider
+                            .insert_hashes(
+                                0..=last_block.number,
+                                last_block.hash(),
+                                *expected_state_root,
+                            )
+                            .map_err(|err| Error::RethError(err.into()))?;
+                    }
+                    _ => return Err(Error::MissingPostState),
                 }
-                _ => return Err(Error::MissingPostState),
-            }
 
-            // Drop the provider without committing to the database.
-            drop(provider);
-        }
+                // Drop the provider without committing to the database.
+                drop(provider);
+                // TODO: replace with `tempdir` usage, so the temp directory is removed
```

Comment:
todo in this PR?

Reply:
https://github.com/paradigmxyz/reth/issues/6824